x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -612,6 +619,21 @@ protected Clock getClock() {
         return components;
     }
 
+    protected void possiblySetElasticPassword(SecurityIndexManager.State previousState, SecurityIndexManager.State currentState) {
+        if (previousState.equals(SecurityIndexManager.State.UNRECOVERED_STATE)
+            && currentState.equals(SecurityIndexManager.State.UNRECOVERED_STATE) == false
+            && securityIndex.get().indexExists() == false
+            && elasticPasswordHash.get() != null) {
+            final ChangePasswordRequest request = new ChangePasswordRequest();
+            request.username("elastic");
+            request.passwordHash(elasticPasswordHash.get().getChars());
+            nativeUsersStore.get().changePassword(request, ActionListener.wrap(
```

Comment:
True, because something else might have created the security index until the second time this node starts.. We could in theory extend the check to be "security index doesn't exist or the doc for the elastic user doesn't exist"